docs/concepts.rst

```diff
@@ -0,0 +1,68 @@
+========
+Concepts
+========
+
+**Kubernetes** is a container orchestrator.
+Over time, Kubernetes has become a standard de facto for the enterprise
+infrastructure management, especially if it based on the microservices.
+
+It provides some basic primitives to orchestrate the application deployments
+on the low level ---such as the pods, jobs, deployments, services, ingresses,
+persistent volumes and volume claims, secrets---
+and allows the Kubernetes cluster to be extended with the arbitrary
+custom resources and custom controllers.
+
+On the top level, it consists of Kubernetes API, through which the users
+talk to Kubernetes, internal storage of the state of the objects (etcd),
+and a collection of the controllers. The command-line tooling (``kubectl``)
+can also be considered as a part of the solution.
+
+----
+
+**Kubernetes controller** is the logic (i.e. the behaviour) behind most
+of the objects, both built-in and added as the extension of Kubernetes.
+For example, the ReplicaSet and Pod creation when a Deployment object
```

Comment:
"Examples of objects are ReplicaSet and Pod creation upon creation of a deployment, rolling version upgrades, ..."